pkg/apis/autoscaling/v1/fleetautoscaler.go

```diff
@@ -423,6 +504,67 @@ func (l *ListPolicy) ValidateListPolicy(fldPath *field.Path) field.ErrorList {
 	return allErrs
 }
 
+// ValidateChainPolicy validates the FleetAutoscaler Chain policy settings.
+func (c *ChainPolicy) ValidateChainPolicy(fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	if !runtime.FeatureEnabled(runtime.FeatureScheduledAutoscaler) {
+		return append(allErrs, field.Forbidden(fldPath, "feature ScheduledAutoscaler must be enabled"))
+	}
+	for i, entry := range c.Items {
+		if entry.ID != "" {
+			for j, otherEntry := range c.Items {
+				// If the entry's id is the same as another entry's id in the chain, append an error
+				if (entry.ID == otherEntry.ID || entry.ID == strconv.Itoa(j)) && i != j {
+					allErrs = append(allErrs, field.Invalid(fldPath.Child("items"), entry.ID, "id of chain entry must be unique"))
+				}
+			}
+		}
+		if entry.Schedule.Between.Start != "" {
```

Comment:
Should we check/validate that the start time is before end time as well ?

Reply:
I was thinking about this, if the user enters an end time that's before the start time then the policy is just never applied.

Reply:
But we don't ouput any error at this moment, right ?

Reply:
Same case with the end date being before the current date.

Reply:
No, we do not.

Reply:
Talked offline w/ Zach he said we should go ahead and validate it.